strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/aql/AqlController.java

```diff
@@ -0,0 +1,94 @@
+package org.carlspring.strongbox.controllers.aql;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+
+import org.carlspring.strongbox.aql.grammar.AqlQueryParser;
+import org.carlspring.strongbox.controllers.BaseController;
+import org.carlspring.strongbox.data.criteria.OQueryTemplate;
+import org.carlspring.strongbox.data.criteria.QueryParserException;
+import org.carlspring.strongbox.data.criteria.Selector;
+import org.carlspring.strongbox.domain.ArtifactEntry;
+import org.carlspring.strongbox.providers.io.RepositoryPath;
+import org.carlspring.strongbox.providers.search.SearchException;
+import org.carlspring.strongbox.services.ArtifactResolutionService;
+import org.carlspring.strongbox.storage.search.SearchResult;
+import org.carlspring.strongbox.storage.search.SearchResults;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+/**
+ * @author sbespalov
+ *
+ */
+@Controller
+@RequestMapping("/api/aql")
+@Api(value = "/api/aql")
+public class AqlController extends BaseController
+{
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    @Inject
+    private ArtifactResolutionService artifactResolutionService;
+
+    @ApiOperation(value = "Used to search for artifacts.", response = SearchResults.class)
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK") })
+    @PreAuthorize("hasAuthority('SEARCH_ARTIFACTS')")
+    @RequestMapping(method = RequestMethod.GET, produces = { MediaType.APPLICATION_JSON_VALUE })
+    public ResponseEntity search(@ApiParam(value = "Search query", required = true) @RequestParam(name = "query", required = true) String query)
+        throws IOException,
+        SearchException
+    {
+        SearchResults result = new SearchResults();
+
+        AqlQueryParser parser = new AqlQueryParser(query);
+        Selector<ArtifactEntry> selector;
+        try
+        {
+            selector = parser.parseQuery();
+        }
+        catch (QueryParserException e)
+        {
+            // TODO: provide response error message
+
+            return ResponseEntity.badRequest().build();
```

Comment:
You should use the [getBadRequestResponseEntity()](https://github.com/strongbox/strongbox/blob/master/strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/BaseController.java#L129) to follow the convention we have introduced in the [wiki](https://github.com/strongbox/strongbox/wiki/How-To-Implement-Spring-Controllers). For the `acceptHeader` just pass `application/json`. I don't think we will be returning anything other than JSON in here for now.

Reply:
ok. thanks @steve-todorov  :)

Reply:
> For the acceptHeader just pass application/json. I don't think we will be returning anything other than JSON in here for now.

I assumed that it should be GET request with path variable parameter.

Reply:
Most of the methods in `BaseController` such as `getBadRequestResponseEntity` depend on the `acceptHeader` to be able to figure out how to return the response to the client (i.e. `text/plain` or `application/json`). However, in this case the controller will only be producing `JSON`, so you can just pass the `application/json` to `acceptHeader` :)